.github/workflows/workflow.yml

```diff
@@ -2,18 +2,22 @@ name: Main workflow
 
 on:
   pull_request:
+    paths-ignore:
+      - '**.md'
   push:
     branches:
       - main
       - releases/*
+    paths-ignore:
+      - '**.md'
 
 jobs:
   build:
     runs-on: ${{ matrix.operating-system }}
     strategy:
       fail-fast: false
       matrix:
-        operating-system: [ubuntu-latest, windows-latest, macOS-latest]
+        operating-system: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04, windows-latest, macOS-latest]
```

Comment:
why do you specify all 3 versions of Ubuntu but only the latest ones for Windows and macOS?

Reply:
We have only one macOS version on GitHub. Maybe it make sense to test `windows-2016` and `windows-2019` images.

Reply:
Actually, I think that there is no need to test on all Ubuntu images. You are verifying how different versions are resolved and then installed, right? If yes, then it shouldn't be different on different OS versions. 
At least, you are not specifying exact OS version in the new issue template, then it means it doesn't matter which exact version is used.

Reply:
I've changed os matrix similarly the other setup tasks, for example [setup-python](https://github.com/actions/setup-python/blob/main/.github/workflows/test-python.yml).

@maxim-lobanov , any thoughts?

Reply:
`macos-latest`, `ubuntu-latest`, `windows-latest` should be enough